### PR TITLE
Add support for wiki edits

### PR DESCRIPTION
**Pull request by jnduli**
## Description

Added api calls to edit wiki data based [on this spec](https://issues.genenetwork.org/issues/edit-rif-metadata)

### Tested:

Example API call:

```
POST http://127.0.0.1:9094/api/wiki/12/edit HTTP/1.1
Content-Type: application/json

{
"symbol": "XXXX",
"pubmed_ids": ["XXXX", "XXXX"],
"species": "mouse",
"comment": "XXXX",
"email": "XXXX",
"web_url": "XXXX",
"initial": "XXXX", 
"categories": ["Behavior", "Biochemistry "],
"reason": "XXXX"
}

```

And here are example responses:
```
# success
HTTP/1.0 200 OK
Content-Type: application/json
Content-Length: 22
Server: Werkzeug/2.0.2 Python/3.10.7
Date: Fri, 23 Aug 2024 08:37:55 GMT


{
  "success": "ok"
}


# Failure
HTTP/1.0 500 INTERNAL SERVER ERROR
Content-Type: application/json
Content-Length: 97
Server: Werkzeug/2.0.2 Python/3.10.7
Date: Fri, 23 Aug 2024 08:38:17 GMT


{
  "error": "Error editting wiki entry, expected 1 species with Name=mouseasdaf but found 0!"
}


```

### CODE REVIEW

**Comment by Alexanderlacuna** [comment]
@jnduli  what is the reason we store the user ip?

**Comment by jnduli** [comment]
> @jnduli what is the reason we store the user ip?

@Alexanderlacuna I'm not too sure. We do have the entries on the database though. I assume it's a security feature where we can revert changes from a malicious actor using their ip.

**Review by BonfaceKilz — COMMENTED** [review]
@jnduli fix the linting errors.  Otherwise, I'm okay merging this.

File: gn3/api/wiki.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    INSERT INTO GeneRIF (Id, versionId, symbol, PubMed_ID, SpeciesID, comment,
+                         email, createtime, user_ip, weburl, initial, reason)
+    VALUES (%(Id)s, %(versionId)s, %(symbol)s, %(PubMed_ID)s, %(SpeciesID)s, %(comment)s, %(email)s, %(createtime)s, %(user_ip)s, %(weburl)s, %(initial)s, %(reason)s)
+    """
+    with db_utils.database_connection(current_app.config["SQL_URI"]) as conn:
+        cursor = conn.cursor()
+        try:
+            category_ids = get_categories_ids(cursor, payload["categories"])
+            species_id = get_species_id(cursor, payload["species"])
+            next_version = get_next_comment_version(cursor, comment_id)
```
Comment (BonfaceKilz) [inline gn3/api/wiki.py]: *suggestion:* @jnduli I would suggest moving the implementation of `get_categories_ids`, `get_species_id`, and `get_next_comment_version` to their own module.  Thtese functions seem re-usable and would make sense to move them to the db module for more re-use.

Comment (jnduli) [reply gn3/api/wiki.py]: thanks. I've moved these to `gn3.db.wiki` and added an extra module `gn3.api.metadata_apis.wiki` to house the wiki apis.

**Comment by jnduli** [comment]
Updated testing information:

```
# request
POST http://127.0.0.1:9094/api/metadata/wiki/12/edit HTTP/1.1
Content-Type: application/json

{
"symbol": "XXXX",
"pubmed_ids": ["XXXX", "XXXX"],
"species": "mouse",
"comment": "XXXX",
"email": "XXXX",
"web_url": "XXXX",
"initial": "XXXX", 
"categories": ["Behavior", "Biochemistry "],
"reason": "XXXX"
}


# response
HTTP/1.0 200 OK
Content-Type: application/json
Content-Length: 22
Server: Werkzeug/2.0.2 Python/3.10.7
Date: Mon, 26 Aug 2024 09:38:46 GMT


{
  "success": "ok"
}


```

**Review by BonfaceKilz — APPROVED** [review]

**Pull request merged**